lib/engine/connection.rb

```diff
@@ -81,17 +81,23 @@ def matches?(connection_hexes)
     def id
       @id ||=
         begin
-          uniq_paths = []
-          junction_map = {}
-
-          # skip over paths that have a junction we've already seen
-          @paths.each do |path|
-            uniq_paths << path if !junction_map[path.a] && !junction_map[path.b]
-            junction_map[path.a] = true if path.a.junction?
-            junction_map[path.b] = true if path.b.junction?
-          end
+          # deal with ambiguous intra-tile path
+          if @paths&.one? && @paths[0].tile.ambiguous_connection?
```

Comment:
why would paths be nil?

Reply:
In initialize: @path = nil

Reply:
then how come the line below doesn't fail?

Reply:
Good point.